common/scanner.h

```diff
@@ -142,6 +143,21 @@ static inline bool external_scanner_scan(void *payload, TSLexer *lexer, const bo
     }
 
     return true;
+  } else if (valid_symbols[FUNCTION_SIGNATURE_SEMICOLON]) {
+    while (lexer->lookahead == '\n') {
+      advance(lexer);
+    }
+
+    // 'iswalpha' checks if a keyword follows, in which case it can only be
+    // function signature (a function declaration would have followed with '{')
+    if (lexer->lookahead == ';' || lexer->lookahead == 0 || iswalpha(lexer->lookahead)) {
+      if (lexer->lookahead == ';' || lexer->lookahead == 0) {
```

Comment:
Unsure if `advance(lexer)` is proper when `lexer->lookahead == 0`. I would think it doesn't matter.